Equilobe.DailyReport.SL/AtlassianWorklogDataService.cs

```diff
@@ -0,0 +1,123 @@
+﻿using Equilobe.DailyReport.DAL;
+using Equilobe.DailyReport.Models.Dashboard;
+using Equilobe.DailyReport.Models.Interfaces;
+using Equilobe.DailyReport.Models.Storage;
+using System;
+using System.Collections.Generic;
+using System.Linq;
+
+namespace Equilobe.DailyReport.SL
+{
+    public class AtlassianWorklogDataService : IAtlassianWorklogDataService
+    {
+        #region IAtlassianWorklogDataService implementation
+        public Dictionary<long, List<AtlassianWorklog>> GetAtlassianWorklogsByUserIds(long instanceId, List<long> usersIds)
+        {
+            using (var db = new ReportsDb())
+            {
+                return db.AtlassianWorklogs
+                    .Where(p => usersIds.Contains(p.AtlassianUserId))
+                    .GroupBy(p => p.AtlassianUserId)
+                    .ToDictionary(p => p.Key, p => p.ToList());
+            }
+        }
+
+        public Dictionary<long, List<DashboardWorklog>> GetLastWorklogsByUsers(List<long> usersIds, ReportContext context, string baseUrl)
+        {
+            using (var db = new ReportsDb())
+            {
+                return db.AtlassianWorklogs
+                    .Where(p => usersIds.Contains(p.AtlassianUserId))
+                    .Where(p => p.StartedAt >= context.BusinessDaysAgo)
+                    .ToList()
```

Comment:
There is no point to convert ToList here.

Reply:
There actually is. I cannot do extension methods or any other methods inside of a linq, until I close the connection to the db, with ToList, in this case. I couldn't have utilized ToOriginalTimeZone extension method to each startedAt otherwise